src/lib/components/Uploader/uploader.component.js

```diff
@@ -0,0 +1,236 @@
+import React, { Component } from "react";
+import { lookup, extension } from "mime-types";
+import auth from "solid-auth-client";
+import { UploadedFiles, SolidError } from "@entities";
+
+type Props = {
+  fileBase: String,
+  limitFiles: number,
+  render: Node,
+  onComplete?: (files: Array<UploadedFiles>) => void,
+  onDrop?: (files: Array<UploadedFiles>) => void,
+  onError?: (error: SolidError) => void,
+  onStart?: () => void,
+};
+
+class Uploader extends Component<Props> {
+  counter: number;
+  positionFile: number;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      dragging: false,
+      files: [],
+      uploadedFiles: []
+    };
+    this.fileInput = React.createRef();
+    this.positionFile = 0;
+  }
+
+  componentDidMount() {
+    window.addEventListener("dragover", (event: React.EventHandler) => {
+      this.overrideEvent(event);
+    });
+    window.addEventListener("drop", (event: React.EventHandler) => {
+      this.overrideEvent(event);
+    });
+  }
+
+  componentDidUpdate(prevProps: Object, prevState: Object) {
+    // When files updates means that we have a new files to upload
```

Comment:
This sentence is a bit confusing, I believe the first "files" refers to the property? If so, let's clarify by mentioning that it's the variable. Also it should either read "we have a new file to upload" or "we have new files to upload".

Reply:
I think this works: Upload file will start when input file change or you drop an file into uploader area.

Let me know what do you think